src/plonk/lookup.rs

```diff
@@ -313,52 +211,41 @@ impl<F: PrimeField> Witness<F> {
     /// calculate the inverse in log derivative formula
     /// h_i := \frac{1}{l_i+r}
     /// g_i := \frac{m_i}{t_i+r}
-    pub fn calc_inverse_terms<C: CurveAffine<ScalarExt = F>>(
-        &self,
-        ss: &Structure<C>,
-        r: F,
-        m: &[F],
-    ) -> (Vec<F>, Vec<F>) {
-        let h = self
-            .l
+    pub fn evaluate_h_g(&self, l: &[F], t: &[F], r: F, m: &[F]) -> (Vec<F>, Vec<F>) {
+        let h = l
             .iter()
             .map(|&l_i| Option::from((l_i + r).invert()).unwrap_or(F::ZERO))
             .collect::<Vec<F>>();
-        let g =
-            ss.t.iter()
-                .zip(m)
-                .map(|(t_i, m_i)| *m_i * Option::from((*t_i + r).invert()).unwrap_or(F::ZERO))
-                .collect::<Vec<F>>();
+        let g = t
+            .iter()
+            .zip(m)
+            .map(|(t_i, m_i)| *m_i * Option::from((*t_i + r).invert()).unwrap_or(F::ZERO))
+            .collect::<Vec<F>>();
         (h, g)
     }
 
     /// check whether the lookup argument is satisfied
     /// the remaining check L(x1,...,xa)=l and T(y1,...,ya)=t
     /// are carried in upper level check
-    pub fn is_sat<C: CurveAffine<ScalarExt = F>>(
-        &self,
-        ss: &Structure<C>,
-        r: F,
-    ) -> Result<(), String> {
-        let m = self.log_derivative_coeffs(ss);
-        let (h, g) = self.calc_inverse_terms(ss, r, &m);
+    pub fn is_sat(&self, l: &[F], t: &[F], r: F) -> Result<(), String> {
```

Comment:
`is_satisfied`

Reply:
I prefer the `is_sat` which is a common term in complexity theory. For example, 3-SAT is a term for boolean satisfiability.